### PR TITLE
Add poetry support to SortingHat

### PR DESCRIPTION
**Pull request by jjmerchante**
This PR:
- Adds the files required to use Poetry for building and installing the package.
- Moves the directories `bin/` and `misc/` inside SortingHat to be used in Poetry as scripts. 
- Changes `setup.py` to use the previous scripts as entry_points based on Perceval.

### CODE REVIEW

**Comment by vchrombie** [comment]
The build is failing because of https://github.com/chaoss/grimoirelab-sortinghat/issues/603 and we have a PR to fix it https://github.com/chaoss/grimoirelab-sortinghat/pull/604

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+keywords = [
+    "development",
+    "grimoirelab"
+]
+
+packages = [
+    { include = "sortinghat" },
+    { include = "tests", format = "sdist" },
+]
+
```
Comment (vchrombie) [inline pyproject.toml:24]: I think we missed this
```
include = [
    { path = "AUTHORS", format = "sdist" },
    { path = "NEWS", format = "sdist" },
    { path = "README.md", format = "sdist" },
]
```
https://github.com/chaoss/grimoirelab-toolkit/blob/a1fed329d25731177105fa094fa22b1029fdac1e/pyproject.toml#L25
https://github.com/chaoss/grimoirelab-toolkit/issues/44

Looks like we missed it in other tools as well

**Comment by sduenas** [comment]
> The build is failing because of #603 and we have a PR to fix it #604

I've just merged the PR. @vchrombie do you think this PR looks good?

**Review by vchrombie — COMMENTED** [review]
I had one minor comment, rest all looks good. I think we need to fix the build first before merging this.

cc @sduenas

**Comment by vchrombie** [comment]
I see the PR https://github.com/chaoss/grimoirelab-sortinghat/pull/604 is merged, it should fix the build failure :+1:

File: pyproject.toml (hunk 1 above)
Comment (vchrombie) [reply pyproject.toml:24]: @sduenas WDYT?

Comment (sduenas) [reply pyproject.toml:24]: Sure, we should add those. Where else did we miss to include those files?

Comment (vchrombie) [reply pyproject.toml:24]: I don't see the configurations in the perceval repo (I just opened a PR https://github.com/chaoss/grimoirelab-perceval/pull/776 for it)
I will check if the other tools also miss this, and will open the PRs accordingly.

**Review by vchrombie — APPROVED** [review]
LGTM! Thanks for working on this jj!

**Pull request closed without merging**